packages/yarnpkg-core/sources/httpUtils.ts

```diff
@@ -86,16 +83,16 @@ export async function request(target: string, body: Body, {configuration, header
 
   const makeHooks = () => ({
     beforeRequest: [
-      (options: NormalizedOptions) => {
+      ((options: NormalizedOptions) => {
         hostname = options.url.hostname;
-      },
+      }) as BeforeRequestHook,
     ],
     beforeRedirect: [
-      (options: NormalizedOptions) => {
+      ((options: NormalizedOptions) => {
         if (options.headers && options.headers.authorization && options.url.hostname !== hostname) {
           delete options.headers.authorization;
         }
-      },
+      }) as unknown as BeforeRedirectHook,
```

Comment:
The `authorization` header leak was fixed in the beta, you can remove this now :)

Reply:
Okay, will try now, thanks!
One minor note, I have noticed that `beforeRedirectHook` uses the different type of `NormalizedOptions` than `NormalizedOptions` exported from top-level got `index.d.ts`, thats why I had to cast to `as unknown as BeforeRedirectHook` to shut up `tsc`.
![image](https://user-images.githubusercontent.com/1259926/79430069-0fe85c00-7fd1-11ea-9214-84391da9c41d.png)